pipeline/pipelines/pipeline_factory.py

```diff
@@ -55,3 +58,36 @@ def _rec_subclasses(cls):
             subclasses.append(subcls)
             subclasses += _rec_subclasses(subcls)
     return subclasses
+
+
+class ConfigLoadingCodeGenerationPipeline(pipeline_base.PipelineBase):
+    """The pipeline which loads the config on the task execution environment.
+
+    This fits well to execute tasks remotely, so local commandline users don't
+    have to set up the pipeline configs.
+    """
+
+    def __init__(self, **kwargs):
+        super(ConfigLoadingCodeGenerationPipeline, self).__init__(**kwargs)
+
+    def do_build_flow(self, **kwargs):
+        flow = linear_flow.Flow('config-loading-codegen')
+        flow.add(config_tasks.ConfigReadTask('ConfigReader', inject=kwargs))
+        pipeline_name = self.get_pipeline_prefix(**kwargs) + 'ClientPipeline'
+        kwargs['suppress_validate_kwargs'] = True
+        pipeline = make_pipeline(pipeline_name, **kwargs)
+        flow.add(pipeline.flow)
+        return flow
+
+    def get_pipeline_prefix(self, **kwargs):
+        language_prefix = kwargs['language']
+        if language_prefix == 'csharp':
+            language_prefix = 'CSharp'
```

Comment:
This is for pipeline name -- most of code generation pipeline is like PythonGapicClientPipeline or JavaGrpcClientPipeline -- <capitalized language name><capitalized client type>ClientPipeline (and an exception is C#, where both C and S are capitalized as CSharpGapicClientPipeline).